ERCS/erc-7432.md

```diff
@@ -323,94 +287,85 @@ The following JSON is an example of [ERC-7432](./eip-7432.md) Metadata:
- `external`.
-* The `revokeRole` and `revokeRoleFrom` functions SHOULD revert if `_revocable` is `false`. They MAY be implemented as `public` or `external`.
-* The `revokeRoleFrom` function MUST revert if the `msg.sender` is not approved to revoke roles on behalf of the `_revoker` or the `_grantee`.
-  It MAY be implemented as `public` or `external`.
+* A role is represented by a `bytes32`, and it's RECOMMENDED to use the `keccak256` of the role's name with its inputs:
+  `bytes32 roleId = keccak256("RoleName(input_type)")`.
+* The `grantRole` function MUST revert if the `expirationDate` is in the past or if the `msg.sender` is not approved to
+  grant roles on behalf of the NFT owner. It MAY be implemented as `public` or `external`.
+* The `revokeRole` function MUST revert if the `msg.sender` is not approved to revoke roles on behalf of the original
+  NFT owner or the `recipient`. It MAY be implemented as `public` or `external`.
+* If `revocable` is false, only the `recipient` can revoke the role. If `revocable` is true, both the `recipient` and
+  the original NFT owner can revoke the role.
+* The `unlockToken` function MUST revert if the `msg.sender` is not approved to unlock the NFT. It MAY be implemented as
+  `public` or `external`.
 * The `setRoleApprovalForAll` function MAY be implemented as `public` or `external`.
-* The `hasNonUniqueRole` function MAY be implemented as `pure` or `view` and SHOULD only confirm if the role assignment 
-  exists and is not expired.
-* The `hasRole` function MAY be implemented as `pure` or `view` and SHOULD check if the assignment exists, is not
-  expired and is the last one granted (does not support simultaneous role assignments).
-* The `roleData` function MAY be implemented as `pure` or `view`, and SHOULD return an empty struct if the role
-  assignment does not exist.
-* The `roleExpirationDate` function MAY be implemented as `pure` or `view`, and SHOULD return zero if the role
-  assignment does not exist.
-* The `isRoleApprovedForAll` function MAY be implemented as `pure` or `view` and SHOULD only return `true` if the
-  `_operator` is approved to grant and revoke roles for all NFTs on behalf of the `_grantor`.
-* Compliant contracts SHOULD support [ERC-165](./eip-165.md).
+* The `ownerOf` function MAY be implemented as `pure` or `view`, and MUST return the address of the original owner of
+  the NFT.
+* The `recipientOf` function MAY be implemented as `pure` or `view`, and MUST return the address of the account that
+  received the role.
+* The `roleData` function MAY be implemented as `pure` or `view`, and MUST return the encoded data passed to the
+  `grantRole` function.
+* The `roleExpirationDate` function MAY be implemented as `pure` or `view`, and MUST return the expiration date of a
+  given role.
+* The `isRoleRevocable` function MAY be implemented as `pure` or `view`, and MUST return whether the role is revocable.
```

Comment:
**suggestion (non-blocking):** Suggestion (non-blocking):** It would be good to add that it should return a default value if the role expires.

Reply:
Only decided to not implement this one. Usually the standards are more flexible